src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using ILLink.Shared;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace ILLink.RoslynAnalyzer
+{
+	[DiagnosticAnalyzer (LanguageNames.CSharp)]
+	public class DynamicallyAccessedMembersAnalyzer : DiagnosticAnalyzer
+	{
+		internal const string DynamicallyAccessedMembers = nameof (DynamicallyAccessedMembers);
+		internal const string DynamicallyAccessedMembersAttribute = nameof (DynamicallyAccessedMembersAttribute);
+
+		static ImmutableArray<DiagnosticDescriptor> GetSupportedDiagnostics ()
+		{
+			var diagDescriptorsList = new List<DiagnosticDescriptor> ();
+			for (int i = 2067; i <= 2091; i++)
```

Comment:
Can we the `DiagnosticId`s here (casted to `int`)? So there are names instead of numbers?

Reply:
So have twenty-somethings lines of `diagDescriptorsList.Add(... (DiagnosticId.<name>))`?

Reply:
I assume he means something like `for (int i = DiagnosticId.Start; i <= DiagnosticId.End; i++)`. Maybe we can add sentinel values equal to the first/last DAMT mismatch enum values.

Reply:
Ah -- that makes sense. I don't know why I thought this referred to the below line.